### PR TITLE
Fix 1574, 1576 and improve title reload

### PR DESCRIPTION
**Pull request by Ghabry**
The title cache change removes one Game_Temp variable and should unbreak "Return To Title" event command.

### CODE REVIEW

**Comment by Ghabry** [comment]
jenkins: test this please

File: src/scene_title.cpp [hunk 1]
```diff
@@ -51,9 +51,9 @@ void Scene_Title::Start() {
...
 
 void Scene_Title::Continue() {
-	if (Game_Temp::restart_title_cache) {
-		// Clear the cache when the game returns to the
-		// title screen e.g. by pressing F12, except the Title Load menu
+	if (restart_title_cache) {
+		// Clear the cache when the game returns to the title screen
+		// e.g. by pressing F12, except the Title Load menu
 		Cache::Clear();
 		AudioSeCache::Clear();
```
Comment (fmatthew5876) [inline src/scene_title.cpp]: Does this logic have to be done in `Scene_Title::Continue()`? Possible to just do it directly in `OnGameLoad()` (rename it) and then call that directly on startup also?

We localized the restart flag which is good. Possible to eliminate it entirely?

Comment (fmatthew5876) [reply src/scene_title.cpp]: Actually I see how this works now. Doing it this way probably makes sense to ensure we don't miss an edge case.

Comment (Ghabry) [reply src/scene_title.cpp]: It should be possible to put some of this tear-down in the Scene_Map deconstructor, this way we don't depend on the Title scene for this but I don't see a simply way to get rid of this boolean for the "Don't restart BGM when returning from Load scene" edge case.

Comment (fmatthew5876) [reply src/scene_title.cpp]: Better to add state to handle it than try to juggle the edge cases. I was originally mistaken. The way you did it here is probably best.

**Comment by Ghabry** [comment]
Fixed the issue found by ENI, forgot to delete the ResetEventCalling added in Clear.

That this was the bug was obvious after I started looking at the diff: https://github.com/EasyRPG/Player/pull/1566/files

**Comment by Ghabry** [comment]
Cherry-picked one of Sormats commits which is known to be correct. But it doesn't fix #1587 so this is something else :/

**Comment by fmatthew5876** [comment]
Are we sure forcing the frame to middle is right? Maybe its a timing issue where the jump doesn't start until you're at the middle frame, not the other way around?

Do you have a test case on that one? I could verify with my inotify script.

**Comment by Ghabry** [comment]
For a proper test you should also remove the old if(IsContinuous) section.

give your character a charset that is significantly different for "frame left/right" compared to "center", add a "On Touch" event which does "Start Jump -> Right -> Right -> End Jump"

Walk on the event multiple times, you will see it.

**Comment by Ghabry** [comment]
Removed sormats commit as there were valid concerns

**Pull request merged**